test/unit/math/prim/fun/complex_schur_decompose_test.cpp

```diff
@@ -0,0 +1,40 @@
+#include <stan/math/prim.hpp>
+#include <test/unit/util.hpp>
+#include <gtest/gtest.h>
+
+template <typename T, typename U>
+void expect_complex_mat_eq(const T& x, const U& y, double tol = 1e-8) {
```

Comment:
This should get pushed up into our AD testing tools and not live in the complex_shur test.   We also have functions to extract real and complex components, and these can just call matrix equality.